releaseNotes/Microsoft.Featuremanagement.Telemetry.ApplicationInsights.md

```diff
@@ -0,0 +1,19 @@
+# Microsoft.FeatureManagement.Telemetry.ApplicationInsights
+
+[Source code][source_code] | [Package (NuGet)][package] | [Samples][samples] | [Product documentation][docs]
+
+## 4.0.0-preview2 - March 7, 2024
+
+### Enhancements
+
+* Added a property to published telemetry named `TargetingId`. This is a more reliable way to connect evaluated feature flags to other telemetry events for a given target. ([#409](https://github.com/microsoft/FeatureManagement-Dotnet/issues/409))
```

Comment:
I like how your other comment puts it. As 
> based on the targeting information during your telemetry analysis

Which calls out that it's no longer some ambient field- but the actual one.

Reply:
> in the previous version using one of Application Insights user id fields.

Sounds like a breaking change.

Reply:
Shouldn't be, correlating via user id is still possible.

Reply:
> Correlating events was already possible in the previous version using one of Application Insights user id fields.

If we did this (add userid to "one of Application Insights user id fields") before and not anymore in the new release. It sounds like a breaking change to me.

Reply:
Yeah that would be, but we don't add it. App Insights package automatically generates and adds user id. In our early examples, we were adding AuthenticatedUserId, but it was added/managed in the customer code and not in the library.

Reply:
> we were adding AuthenticatedUserId, but it was added/managed in the customer code and not in the library

Got it. This means we didn't provide a solution out of the box before and now we do. Customers must do it in their application code before this release. Yes, correlating via user id was already possible, but for the sake of the argument :), the whole feature management was possible by user code completely. The release notes are about what we provide out-of-the box regardless of whether something was possible or not by the user code.

> I think this change is more about decoupling the id from Application Insights / not locking ourselves to users...

This is true from our perspective. From the customer perspective, it's all about correlating events. How to do it? It's an implementation detail.

Reply:
True, but just to note again. Application Insights generates `UserId` out-of-the-box which can be used to correlate without custom customer code. Its limitation is hard refreshing or switching apps wouldn't track the user. Our example used `AuthenticatedUserId` because we wanted to get around that. We could say `TargetingId` is the new out-of-the-box way to get around it.

Reply:
Updated to your original suggestion:

> Added a TargetingId property to the feature evaluation events sent to Application Insights. The TargetingId is the identifier of a targeted user during feature evaluation. This new property allows you to correlate feature evaluation events with other telemetry data your application sends to Application Insights, as long as they share the same TargetingId.